### PR TITLE
[Android] Replace current file browser

### PR DESCRIPTION
**Pull request by mahdihijazi**
1. Allow users to pick games dircetory from external storgae.
2. Better UX experince to distinguish between selecting a directory or
a game. The later is needed when we implement change disk for android.
3. fix refreshing the games list after adding a directory in mobile devices.

This PR & [PR#6270](https://github.com/dolphin-emu/dolphin/pull/6270) will make introducing the change disc feature for the Android client easier. Should be my next thing to do.

### CODE REVIEW

**Comment by degasus** [comment]
Do you know how the external storages are mounted for the native app? Does Android uses fuse now?

And as this is a visual change, may I ask for screenshots? ;)

**Comment by mahdihijazi** [comment]
@degasus forgive my lack of knowledge since this is the first time I hear of Fuse and the links might be not relative but from a quick look it seems Android has this for ages:
https://www.quora.com/Does-the-Android-kernel-support-the-FUSE-API-natively
https://www.xda-developers.com/diving-into-sdcardfs-how-googles-fuse-replacement-will-reduce-io-overhead/

form what I understood, FUSE should allow the native code to access external storage without any extra effort which is what happened in this PR since it is working on Shield TV without touching any native code.

**Mobile**
Mobile

**TV**
TV

**External Storage Access**
External Storage Access

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/FileBrowserHelper.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+        // Set these depending on your use case. These are the defaults.
+        i.putExtra(FilePickerActivity.EXTRA_ALLOW_MULTIPLE, false);
+        i.putExtra(FilePickerActivity.EXTRA_ALLOW_CREATE_DIR, false);
+        i.putExtra(FilePickerActivity.EXTRA_MODE, FilePickerActivity.MODE_DIR);
+
+        // Configure initial directory by specifying a String.
+        // You could specify a String like "/storage/emulated/0/", but that can
+        // dangerous. Always use Android's API calls to get paths to the SD-card or
+        // internal memory.
```
Comment (JosJuice) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/FileBrowserHelper.java]: The comments in this method look like they're from example code. I don't think they're appropriate to keep.

File: Source/Android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -74,6 +78,16 @@
             android:exported="false">
         </provider>
 
+        <provider
+            android:name="android.support.v4.content.FileProvider"
+            android:authorities="${applicationId}.filesprovider"
```
Comment (degasus) [inline Source/Android/app/src/main/AndroidManifest.xml]: I think this breaks debug builds. They have a postfix on the application id. But we may just drop this postfix instead...

Comment (mahdihijazi) [reply Source/Android/app/src/main/AndroidManifest.xml]: hmm..probably you are right but I wonder why I didn't see it breaking during the development 🤔

Comment (mahdihijazi) [reply Source/Android/app/src/main/AndroidManifest.xml]: Did you receive any bug report regarding this? What make you come back and check it :)

**Pull request merged**